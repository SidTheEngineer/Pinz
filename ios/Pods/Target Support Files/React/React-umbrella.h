#ifdef __OBJC__
#import <UIKit/UIKit.h>
#endif

#import "RCTAssert.h"
#import "RCTBridge+Private.h"
#import "RCTBridge.h"
#import "RCTBridgeDelegate.h"
#import "RCTBridgeMethod.h"
#import "RCTBridgeModule.h"
#import "RCTBundleURLProvider.h"
#import "RCTConvert.h"
#import "RCTDefines.h"
#import "RCTDisplayLink.h"
#import "RCTErrorCustomizer.h"
#import "RCTErrorInfo.h"
#import "RCTEventDispatcher.h"
#import "RCTFrameUpdate.h"
#import "RCTImageSource.h"
#import "RCTInvalidating.h"
#import "RCTJavaScriptExecutor.h"
#import "RCTJavaScriptLoader.h"
#import "RCTJSStackFrame.h"
#import "RCTKeyCommands.h"
#import "RCTLog.h"
#import "RCTModuleData.h"
#import "RCTModuleMethod.h"
#import "RCTMultipartDataTask.h"
#import "RCTMultipartStreamReader.h"
#import "RCTNullability.h"
#import "RCTParserUtils.h"
#import "RCTPerformanceLogger.h"
#import "RCTRootView.h"
#import "RCTRootViewDelegate.h"
#import "RCTRootViewInternal.h"
#import "RCTTouchEvent.h"
#import "RCTTouchHandler.h"
#import "RCTURLRequestDelegate.h"
#import "RCTURLRequestHandler.h"
#import "RCTUtils.h"
#import "RCTWebSocketProxy.h"
#import "RCTWebSocketProxyDelegate.h"
#import "CSSLayout.h"
#import "CSSMacros.h"
#import "CSSNodeList.h"
#import "RCTJSCErrorHandling.h"
#import "RCTJSCExecutor.h"
#import "RCTJSCWrapper.h"
#import "RCTAccessibilityManager.h"
#import "RCTAlertManager.h"
#import "RCTAppState.h"
#import "RCTAsyncLocalStorage.h"
#import "RCTClipboard.h"
#import "RCTDevLoadingView.h"
#import "RCTDevMenu.h"
#import "RCTEventEmitter.h"
#import "RCTExceptionsManager.h"
#import "RCTI18nManager.h"
#import "RCTI18nUtil.h"
#import "RCTKeyboardObserver.h"
#import "RCTRedBox.h"
#import "RCTSourceCode.h"
#import "RCTStatusBarManager.h"
#import "RCTTiming.h"
#import "RCTUIManager.h"
#import "RCTFPSGraph.h"
#import "RCTJSCProfiler.h"
#import "RCTMacros.h"
#import "RCTProfile.h"
#import "RCTActivityIndicatorView.h"
#import "RCTActivityIndicatorViewManager.h"
#import "RCTAnimationType.h"
#import "RCTAutoInsetsProtocol.h"
#import "RCTBorderDrawing.h"
#import "RCTBorderStyle.h"
#import "RCTComponent.h"
#import "RCTComponentData.h"
#import "RCTConvert+CoreLocation.h"
#import "RCTConvert+MapKit.h"
#import "RCTDatePicker.h"
#import "RCTDatePickerManager.h"
#import "RCTFont.h"
#import "RCTMap.h"
#import "RCTMapAnnotation.h"
#import "RCTMapManager.h"
#import "RCTMapOverlay.h"
#import "RCTModalHostView.h"
#import "RCTModalHostViewController.h"
#import "RCTModalHostViewManager.h"
#import "RCTNavigator.h"
#import "RCTNavigatorManager.h"
#import "RCTNavItem.h"
#import "RCTNavItemManager.h"
#import "RCTPicker.h"
#import "RCTPickerManager.h"
#import "RCTPointerEvents.h"
#import "RCTProgressViewManager.h"
#import "RCTRefreshControl.h"
#import "RCTRefreshControlManager.h"
#import "RCTRootShadowView.h"
#import "RCTScrollableProtocol.h"
#import "RCTScrollView.h"
#import "RCTScrollViewManager.h"
#import "RCTSegmentedControl.h"
#import "RCTSegmentedControlManager.h"
#import "RCTShadowView.h"
#import "RCTSlider.h"
#import "RCTSliderManager.h"
#import "RCTSwitch.h"
#import "RCTSwitchManager.h"
#import "RCTTabBar.h"
#import "RCTTabBarItem.h"
#import "RCTTabBarItemManager.h"
#import "RCTTabBarManager.h"
#import "RCTTextDecorationLineType.h"
#import "RCTView.h"
#import "RCTViewControllerProtocol.h"
#import "RCTViewManager.h"
#import "RCTWebView.h"
#import "RCTWebViewManager.h"
#import "RCTWrapperViewController.h"
#import "UIView+Private.h"
#import "UIView+React.h"
#import "RCTActionSheetManager.h"
#import "RCTLocationObserver.h"
#import "RCTGIFImageDecoder.h"
#import "RCTImageBlurUtils.h"
#import "RCTImageCache.h"
#import "RCTImageEditingManager.h"
#import "RCTImageLoader.h"
#import "RCTImageStoreManager.h"
#import "RCTImageUtils.h"
#import "RCTImageView.h"
#import "RCTImageViewManager.h"
#import "RCTLocalAssetImageLoader.h"
#import "RCTResizeMode.h"
#import "RCTLinkingManager.h"
#import "RCTDataRequestHandler.h"
#import "RCTFileRequestHandler.h"
#import "RCTHTTPRequestHandler.h"
#import "RCTNetInfo.h"
#import "RCTNetworking.h"
#import "RCTNetworkTask.h"
#import "RCTSettingsManager.h"
#import "RCTRawTextManager.h"
#import "RCTShadowRawText.h"
#import "RCTShadowText.h"
#import "RCTText.h"
#import "RCTTextField.h"
#import "RCTTextFieldManager.h"
#import "RCTTextManager.h"
#import "RCTTextSelection.h"
#import "RCTTextView.h"
#import "RCTTextViewManager.h"
#import "RCTVibration.h"
#import "RCTSRWebSocket.h"
#import "RCTWebSocketExecutor.h"
#import "RCTWebSocketManager.h"
#import "RCTWebSocketModule.h"

FOUNDATION_EXPORT double ReactVersionNumber;
FOUNDATION_EXPORT const unsigned char ReactVersionString[];

