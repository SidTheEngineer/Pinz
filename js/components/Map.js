import React from 'react';
import { StyleSheet } from 'react-native'
import MapView from 'react-native-maps';

const STUDENT_UNION = {
  lat: 28.601660,
  lng: -81.200788
}

const styles = StyleSheet.create({
  map: {
    ...StyleSheet.absoluteFillObject
  }
});

const Map = () => (
  <MapView
    initialRegion={{
      latitude: STUDENT_UNION.lat,
      longitude: STUDENT_UNION.lng,
      latitudeDelta: 0.0922,
      longitudeDelta: 0.0421,
    }}
    style={styles.map}
    provider="google"
  />
);

export default Map
