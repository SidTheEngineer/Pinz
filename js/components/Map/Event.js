import React from 'react';
import MapView from 'react-native-maps';

import MAP from '../../constant';
import active from './assets/active.png';
import place from './assets/place.png';

// Supply coordinates of event based on name of location.
const getEventCoords = (event) => {
  const landmarks = MAP.LANDMARKS.filter(
    landmark => landmark.NAME === event.location
  );
  console.log(event.title);
  return landmarks[0].REGION;
};

// Supply a custom marker image based on category of event.
const getEventImage = (event) => {
  switch (event.category) {
    case 'Sports':
      return active;
    default:
      return place;
  }
};

const Event = props => (
  <MapView.Marker
    key={props.details.event_id}
    coordinate={getEventCoords(props.details)}
    image={getEventImage(props.details)}
    // draggable
  />
);

Event.propTypes = {
  details: React.PropTypes.shape({
    event_id: React.PropTypes.string
  })
};

export default Event;
