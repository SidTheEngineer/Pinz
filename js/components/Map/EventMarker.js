import React from 'react';
import MapView from 'react-native-maps';
import { Text, StyleSheet } from 'react-native';

import MAP, { COLORS } from '../../constant';
import active from './assets/active.png';
import place from './assets/place.png';

const styles = StyleSheet.create({
  calloutContainer: {
    height: 25,
    width: 250,
    backgroundColor: COLORS.RED,
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    fontWeight: 'bold',
  },
  text: {
    color: 'white',
    fontSize: 12,
  }
});

// Supply coordinates of event based on name of location.
const getEventCoords = (event) => {
  const landmarks = MAP.LANDMARKS.filter(
    landmark => landmark.NAME === event.location
  );
  return landmarks[0].REGION;
};

// Supply a custom marker image based on category of event.
const getEventImage = (event) => {
  switch (event.category) {
    case 'Sports':
      return active;
    default:
      return place;
  }
};

const EventMarker = props => (
  <MapView.Marker
    key={props.details.event_id}
    coordinate={getEventCoords(props.details)}
    image={getEventImage(props.details)}
  >
    <MapView.Callout tooltip style={styles.calloutContainer}>
      <Text style={[styles.text, styles.title]}>
        {props.details.title.substring(0, 40)}
      </Text>
    </MapView.Callout>
  </MapView.Marker>
);

EventMarker.propTypes = {
  details: React.PropTypes.shape({
    event_id: React.PropTypes.string,
    title: React.PropTypes.string,
    description: React.PropTypes.string
  })
};

export default EventMarker;
