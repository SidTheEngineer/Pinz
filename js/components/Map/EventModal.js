import React from 'react';
import { Modal, StyleSheet, View } from 'react-native';

const styles = StyleSheet.create({
  modalView: {
    height: 500,
    width: 300,
    backgroundColor: 'white'
  }
});

const EventModal = props => (
  <Modal
    visible={props.map.modalVisibility}
    transparent
    animationType={'fade'}
    onRequestClose={() => props.mapActions.showEventModal(false)}
  >
    <View style={styles.modalView} />
  </Modal>
);

EventModal.propTypes = {
  map: React.PropTypes.shape({
    modalVisibility: React.PropTypes.bool.isRequired
  }),
  mapActions: React.PropTypes.objectOf(React.PropTypes.func)
};

export default EventModal;
