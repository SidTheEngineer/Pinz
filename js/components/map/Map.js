import React from 'react';
import { StyleSheet } from 'react-native';
import MapView from 'react-native-maps';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

const styles = StyleSheet.create({
  map: {
    ...StyleSheet.absoluteFillObject
  }
});

const Map = props => (
  <MapView
    region={props.map.initialRegion}
    style={styles.map}
    provider="google"
    scrollEnabled={false}
    zoomEnabled={false}
  />
);

Map.propTypes = {
  map: React.PropTypes.objectOf(React.PropTypes.object),
  initialRegion: React.PropTypes.shape({
    latitude: React.PropTypes.number,
    longitude: React.PropTypes.number,
    latitudeDelta: React.PropTypes.number,
    longitudeDelta: React.PropTypes.number
  })
};

const mapStateToProps = state => state;

// TODO: Bind future action creators here.
const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({}, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(Map);
