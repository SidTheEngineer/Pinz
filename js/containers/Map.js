import React, { Component } from 'react';
import { StyleSheet, StatusBar } from 'react-native';
import MapView from 'react-native-maps';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import ViewContainer from '../components/ViewContainer';

const styles = StyleSheet.create({
  map: {
    ...StyleSheet.absoluteFillObject
  }
});

class Map extends Component {

  constructor(props) {
    super(props);

    this.handleZoomRestriction = this.handleZoomRestriction.bind(this);
  }

  handleZoomRestriction() {
    console.log(this);
  }

  render() {
    return (
      <ViewContainer>
        <StatusBar
          hidden
        />
        <MapView
          region={this.props.map.initialRegion}
          style={styles.map}
          provider="google"
          scrollEnabled={false}
          rotateEnabled={false}
          pitchEnabled={false}
          showsCompass={false}
          loadingEnabled
<<<<<<< HEAD
          onRegionChange={this.handleZoomRestriction}
=======
          onRegionChange={this.handleZoomRestriction()}
>>>>>>> 9c8baa72135442c33d49c46317acadfb0c73abca
        />
      </ViewContainer>
    );
  }
}

Map.propTypes = {
  map: React.PropTypes.objectOf(React.PropTypes.object),
  initialRegion: React.PropTypes.shape({
    latitude: React.PropTypes.number,
    longitude: React.PropTypes.number,
    latitudeDelta: React.PropTypes.number,
    longitudeDelta: React.PropTypes.number
  })
};

const mapStateToProps = state => state;

// TODO: Bind future action creators here.
const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({}, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(Map);
