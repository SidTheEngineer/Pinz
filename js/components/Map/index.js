import React, { Component } from 'react';
import { View, StyleSheet, StatusBar, Platform } from 'react-native';
import MapView from 'react-native-maps';

import ViewContainer from '../ViewContainer';
import MAP, { COLORS } from '../../constant';
import EventMarker from './EventMarker';
import EventModal from './EventModal';

const styles = StyleSheet.create({
  map: { flex: 5 },
  coordinates: { flex: 1 },
  statusBarContainer: {
    height: Platform.OS === 'ios' ? 20 : 0,
    backgroundColor: COLORS.LIGHT_GREEN
  }
});

class Map extends Component {

  static isInBounds(region) {
    return (
         (region.latitudeDelta < MAP.BOUNDS.MIN_ZOOM)
      && (region.longitude < MAP.BOUNDS.RIGHT_BOUNDS)
      && (region.latitude < MAP.BOUNDS.TOP_BOUNDS)
      && (region.longitude > MAP.BOUNDS.LEFT_BOUNDS)
      && (region.latitude > MAP.BOUNDS.BOTTOM_BOUNDS)
    );
  }

  constructor(props) {
    super(props);

    this.state = {
      region: MAP.INITIAL_REGION,
      events: [],
      modalVisible: false
    };

    this.onRegionChange = this.onRegionChange.bind(this);
    this.listEventMarkers = this.listEventMarkers.bind(this);
    this.setModalVisibility = this.setModalVisibility.bind(this);
  }

  componentDidMount() {
    this.props.mapActions.fetchInitialEvents();
  }

  onRegionChange(region) {
    this.setState({ region });

    if (!Map.isInBounds(region)) {
      this.setState({ region: MAP.INITIAL_REGION });
    }
  }

  setModalVisibility(visible) {
    this.setState({ modalVisible: visible });
  }

  listEventMarkers() {
    return this.props.map.events.map(event => (
      <EventMarker
        key={event.event_id}
        details={event}
        loadModal={() => this.props.mapActions.showEventModal(true)}
      />
    ));
  }

  render() {
    return (
      <ViewContainer>
        <View style={styles.statusBarContainer}>
          <StatusBar
            barStyle="light-content"
            backgroundColor={COLORS.LIGHT_GREEN}
          />
        </View>
        <MapView
          region={this.state.region}
          style={styles.map}
          provider="google"
          rotateEnabled={false}
          pitchEnabled={false}
          showsCompass={false}
          loadingEnabled
          moveOnMarkerPress={false}
          onRegionChange={this.onRegionChange}
        >
          {
            this.listEventMarkers()
          }
        </MapView>
        {/* <Text style={styles.coordinates}>
          Latitude: {this.state.region.latitude}{'\n'}
          Longitude: {this.state.region.longitude}{'\n'}
          Latitude Delta: {this.state.region.latitudeDelta}{'\n'}
          Longitude Delta: {this.state.region.longitudeDelta}{'\n'}
        </Text> */}
        <EventModal {...this.props} />
      </ViewContainer>
    );
  }
}

Map.propTypes = {
  map: React.PropTypes.shape({
    events: React.PropTypes.oneOfType([
      React.PropTypes.arrayOf(React.PropTypes.object),
      React.PropTypes.object
    ]),
    modalVisibility: React.PropTypes.bool.isRequired
  }),
  mapActions: React.PropTypes.objectOf(React.PropTypes.func)
};

export default Map;
