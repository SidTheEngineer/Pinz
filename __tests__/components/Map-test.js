import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Map from '../../js/components/Map';
import MAP from '../../js/constant';

jest.mock('react-native-maps', () => 'MapView');
jest.mock('../../js/components/Map/EventMarker', () => 'EventMarker');

const props = {
  map: {
    modalVisibility: false,
    modalDetails: null,
    events: [
      {
        event_id: '12345',
        id: '12345',
        title: 'This is a test event',
        description: 'test test test',
        location: 'Location 123',
        starts: 'midnight',
        ends: 'midnight',
        category: 'Sports'
      }
    ]
  },
  mapActions: {
    fetchInitialEvents: jest.fn()
  }
};

expect.extend({
  // Takes in an area where the map would consider out of bounds.
  // Returns true if out of bounds.
  toBeOutOfBounds(failRegion, passRegion) {
    const pass = !Map.isInBounds(Object.assign(
      {},
      passRegion,
      failRegion
    ));

    const message =
      () => `expected ${failRegion} ${pass ? 'not' : ''} to be true`;

    return { message, pass };
  }
});

describe('Map', () => {
  it('renders correctly', () => {
    const map = renderer.create(
      <Map {...props} />
    ).toJSON();
    expect(map).toMatchSnapshot();
  });

  it('checks for out of bounds', () => {
    // Min zoom check.
    expect({ latitudeDelta: 0.026 }).toBeOutOfBounds(MAP.INITIAL_REGION);

    // Right bounds check.
    expect({ longitude: -81 }).toBeOutOfBounds(MAP.INITIAL_REGION);

    // Top bounds check.
    expect({ latitude: 28.7 }).toBeOutOfBounds(MAP.INITIAL_REGION);

    // Left bounds check.
    expect({ longitude: -81 }).toBeOutOfBounds(MAP.INITIAL_REGION);

    // Bottom bounds check.
    expect({ latitude: 28 }).toBeOutOfBounds(MAP.INITIAL_REGION);
  });
});
