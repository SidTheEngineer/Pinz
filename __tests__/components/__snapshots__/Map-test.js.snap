exports[`Map renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
      Object {},
    ]
  }>
  <MapView
    loadingEnabled={true}
    onRegionChange={[Function]}
    pitchEnabled={false}
    provider="google"
    region={
      Object {
        "latitude": 28.60166,
        "latitudeDelta": 0.0206,
        "longitude": -81.200788,
        "longitudeDelta": 0.014,
      }
    }
    rotateEnabled={false}
    showsCompass={false}
    style={
      Object {
        "flex": 5,
      }
    } />
</View>
`;
